libs/s25main/network/GameClient.cpp

```diff
@@ -758,7 +758,7 @@ bool GameClient::OnGameMessage(const GameMessage_Chat& msg)
 {
     if(msg.destination == CD_SYSTEM)
     {
-        SystemChat(msg.text, msg.player);
+        SystemChat(msg.text, (msg.player < game->world_.GetNumPlayers()) ? msg.player : GetPlayerId());
```

Comment:
shouldnt this the "system-player"? or do I understand correctly that this is then then the local player?

cant we simply write `SystemChat(msg.text)` here?

Reply:
We have 1 example were this is used: When a player saves the game (which will cause a lag) he sends a message to be displayed to all players. In that case `msg.player` is set to that player and must be used. However we can also just send a message which should be displayed as `SystemChat(msg.text)` which is equivalent to `SystemChat(msg.text, GetPlayerId());` and done by setting `msg.player=0xFF`.

Kinda unusual but that's what it is currently.